src/AzureSqlActionHelper.ts

```diff
@@ -116,28 +125,39 @@ export default class AzureSqlActionHelper {
-     *  2. SSDT (SQL Server Data Tools) installs it in location VS Install Directory/Common7/IDE/Extensions/Microsoft/SQLDB/DAC/{SqlVersion}
+     * SqlPackage.exe can be installed in four ways:
+     *  0. as a global dotnet tool
+     *  1. SQL Server Management Studio (SSMS) used to install it to in location C:/Program Files/Microsoft SQL Server/{SqlVersopn}/DAC/bin/SqlPackage.exe' (REMOVE in the future)
+     *  2. the Dac Framework MSI installs it in location C:/Program Files/Microsoft SQL Server/{SqlVersopn}/DAC/bin/SqlPackage.exe'
+     *  3. SSDT (SQL Server Data Tools) installs it in location VS Install Directory/Common7/IDE/Extensions/Microsoft/SQLDB/DAC/{SqlVersion}
      * 
      *  This method finds the location of SqlPackage.exe from both the location and return the highest version of SqlPackage.exe
      */
     private static async _getSqlPackageExecutablePath(): Promise<string> {
-        core.debug('Getting location of SQLPackage.exe');
+        core.debug('Getting location of SqlPackage');
+
+        let sqlPackagePathInstalledWithDotnetTool = await this._getSqlPackageExeInstalledDotnetTool();
+        core.debug(`SqlPackage (installed with dotnet tool) found at location: ${sqlPackagePathInstalledWithDotnetTool[0]}, version ${sqlPackagePathInstalledWithDotnetTool[1]}`);
 
         let sqlPackagePathInstalledWithSSMS = await this._getSqlPackageInstalledWithSSMS();
-        core.debug(sqlPackagePathInstalledWithSSMS.toString());
+        core.debug(`SqlPackage (installed with SSMS) found at location: ${sqlPackagePathInstalledWithSSMS[0]}, version ${sqlPackagePathInstalledWithSSMS[1]}`);
 
         let sqlPackagePathInstalledWithDacMsi = await this._getSqlPackageInstalledWithDacMsi();
-        core.debug(sqlPackagePathInstalledWithDacMsi.toString());
+        core.debug(`SqlPackage (installed with DacFramework) found at location: ${sqlPackagePathInstalledWithDacMsi[0]}, version ${sqlPackagePathInstalledWithDacMsi[1]}`);
 
         let sqlPackagePatInstalledWithSSDT = await this._getSqlPackageInstalledWithSSDT();
-        core.debug(sqlPackagePatInstalledWithSSDT.toString());
-        
-        let maximumVersion = Math.max(sqlPackagePathInstalledWithSSMS[1], sqlPackagePathInstalledWithDacMsi[1], sqlPackagePatInstalledWithSSDT[1]);
+        core.debug(`SqlPackage (installed with SSDT) found at location: ${sqlPackagePatInstalledWithSSDT[0]}, version ${sqlPackagePatInstalledWithSSDT[1]}`);
+
 
+        // sort the versions in descending order and return the first path
+        let sqlPackageVersions = [sqlPackagePathInstalledWithDotnetTool[1], sqlPackagePathInstalledWithSSMS[1], sqlPackagePathInstalledWithDacMsi[1], sqlPackagePatInstalledWithSSDT[1]];
+        semver.rsort(sqlPackageVersions);
+        let maximumVersion = sqlPackageVersions[0];
         let sqlPackagePath = '';
-        if (maximumVersion === sqlPackagePathInstalledWithSSMS[1]) {
+        if (maximumVersion === sqlPackagePathInstalledWithDotnetTool[1]) {
```

Comment:
This should really be a `for` or `foreach` loop at this point to match a version with a path.

Reply:
ripped out at part of moving to an interface